packages/tracing/src/transaction.ts

```diff
@@ -119,4 +119,30 @@ export class Transaction extends SpanClass implements TransactionInterface {
 
     return this._hub.captureEvent(transaction);
   }
+
+  /**
+   * @inheritDoc
+   */
+  public toContext(): TransactionContext {
+    const spanContext = super.toContext();
+
+    return dropUndefinedKeys({
+      ...spanContext,
+      name: this.name,
+      trimEnd: this._trimEnd,
+    });
+  }
+
+  /**
+   * @inheritDoc
+   */
+  public updateWithContext(transactionContext: TransactionContext): TransactionInterface {
```

Comment:
Is there a reason this is returning a `TransactionInterface` rather than a `Transaction` (as it does in the type declaration)? 

Or perhaps a more basic question, does it (and the corresponding method on `Span`) need to return anything? Unless you have a strong need to chain it, IMHO it's more obvious that it modifies the object in place if it doesn't return anything. Up to you, though.

Reply:
Regarding the `TransactionInterface`, in this file the `Transaction` from `@sentry/types` is imported `as TransactionInterface`.

And yeah I don't think it does need to return anything but I noticed all the setter methods returned `this`, and now that I noticed it the type declaration returns `this` and not `Span`/`Transaction` which I guess makes it more clear it's in place?

Reply:
Ah, yeah, I see.

In that case, yes, I think using `this` as a return type (both here and in the types file) is a better option.

Reply:
Done, thanks @lobsterkatie!